src/pages/household/input/VariableEditor.jsx

```diff
@@ -367,20 +365,9 @@ export function addVariable(householdInput, variable, entityPlural, year) {
       possibleEntities.forEach((entity) => {
         // If the variable isn't already stored in the situation...
         if (!(variable.name in householdInput[entityPlural][entity])) {
```

Comment:
```suggestion
        if (!(variable.name in householdInput?.[entityPlural]?.[entity])) {
```

Just in case there are more profound errors passed by some other part of the app, I'd suggest this to prevent runtime app failure

Reply:
This optional chaining is unnecessary due to line 363 and where `VaraibleEditor` is used in `HouseholdPage`.

I understand the need for caution, but using optional chaining everywhere is not the answer. I once worked on a legacy codebase where every line was surrounded by 
```
if (x != null) ...
```
and this style was enforced during code reviews. Mandates like this lead to ugly conditional code, IMO. The preconditions, postconditions, and invariants of functions/components should be used to avoid errors including accessing empty objects. Of course, this process is error-prone -- that is why we need good reviews and tests.

Even if we were to use optional chaining here, which is unnecessary, we would need
```
if (!(variable.name in (householdInput?.[entityPlural]?.[entity] || {}))) {
```
instead of the suggested change.

Reply:
Fair enough, and admittedly, the portion of the code this would touch is mostly controlled by the back end, anyway. I'll merge as-is.